src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -1296,9 +1334,36 @@ async def _handle_async_llm_error(
+                    await asyncio.sleep(classification.backoff_seconds)
+                
+                # Temporarily override the model for this call
+                original_llm = self.llm
+                try:
+                    self.llm = next_model
+                    return await self._execute_unified_achat_completion(
+                        messages, temperature, tools, stream,
+                        reasoning_steps, task_name, task_description, task_id, response_format,
+                        stream_callback, emit_events
+                    )
+                except Exception as retry_error:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)